htlcswitch/hop/error_encryptor.go

```diff
@@ -27,9 +27,9 @@ const (
 	EncrypterTypeMock = 2
 )
 
-// ErrorEncrypterExtracter defines a function signature that extracts an
-// ErrorEncrypter from an sphinx OnionPacket.
-type ErrorEncrypterExtracter func(*btcec.PublicKey) (ErrorEncrypter,
+// SharedSecretGenerator defines a function signature that extracts a shared
+// secret from an sphinx OnionPacket.
+type SharedSecretGenerator func(*btcec.PublicKey) (sphinx.Hash256,
```

Comment:
Could you give me the tl;dr for why we're changing this? Getting lost in the various layers of interfaces.

Reply:
The problem is that without this refactor commit, `OnionProcessor` would need to know what kind of sphinx error encrypter to create in https://github.com/lightningnetwork/lnd/blob/17d7e84b26e3242e7f8447c8a40f0a12ce17400b/htlcswitch/hop/iterator.go#L403, where the scope at that point is really just the onion and the shared secret.

The circuit map calls into this method at https://github.com/lightningnetwork/lnd/blob/17d7e84b26e3242e7f8447c8a40f0a12ce17400b/htlcswitch/circuit_map.go#L636 after a restart. It might be necessary to get a attributable error encrypter from `OnionProcessor` then.

I've spent a crazy amount of time getting lost in the interfaces too, and also prototyped multiple approaches to finally settle on what I've got currently.

Will update the commit message with this reasoning.